`node:` imports aren't supported in Node.js v14

The [rule.js](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/rules/utils/rule.js#L2-L3) file uses `node:` imports which are supported since Node.js v16, but `eslint-plugin-unicorn` says it supports v14 and up.

The original error message:

```
Error: Failed to load plugin 'unicorn' declared in '.eslintrc.json': Cannot find module 'node:path'
Require stack:
- eslint-plugin-unicorn@43.0.2_eslint@8.20.0/node_modules/eslint-plugin-unicorn/rules/utils/rule.js
- eslint-plugin-unicorn@43.0.2_eslint@8.20.0/node_modules/eslint-plugin-unicorn/index.js
- @eslint+eslintrc@1.3.0/node_modules/@eslint/eslintrc/dist/eslintrc.cjs
```

The issue occurs after upgrading from 42.0.0 to 43.0.2.

It's a bit confusing that the tests are passing despite that they're running with v14, too.

--- a/docs/rules/prefer-node-protocol.md
+++ b/docs/rules/prefer-node-protocol.md
@@ -9,6 +9,12 @@
 
 When importing builtin modules, it's better to use the [`node:` protocol](https://nodejs.org/api/esm.html#node-imports) as it makes it perfectly clear that the package is a Node.js builtin module.
 
+Note that Node.js support for this feature began in:
+
+> v16.0.0, v14.18.0 (`require()`)
+>
+> v14.13.1, v12.20.0 (`import`)
+
 ## Fail
 
 ```js

